### DISCUSSION TITLE
how to validate a form if inputs

### DISCUSSION THREAD

**Discussion by m1x-developer** [mantinedev/mantine#6426] [Q&A]
how to validate a form if inputs come from the backend in the format {key:'key',value:'value'}

after the response from the back, I initialize the form via form.initialize, after which there must be validation for the number of characters

![image](https://github.com/mantinedev/mantine/assets/83704092/deaa2b73-d965-4a4f-a979-92ea0f98b26f)

**Comment by rtivital** [comment]
What does not work from your screenshot? Is it js or ts error?

**Comment by m1x-developer** [comment]
ts error 
![image](https://github.com/mantinedev/mantine/assets/83704092/eaff24d1-7541-403d-a24d-340792d43378)

**Reply by rtivital** [reply]
`validate` function must return `Record<string, React.ReactNode>` type, your function returns an array of objects.

**Comment by m1x-developer** [comment]
Validation still doesn't work :(

![image](https://github.com/mantinedev/mantine/assets/83704092/0d133fe1-a8ca-44f9-a9f7-72d113e9c30f)
![image](https://github.com/mantinedev/mantine/assets/83704092/e7cf29ce-f13a-4e62-bc30-8909e1c7f35c)

**Reply by rtivital** [reply]
https://stackoverflow.com/a/59321797/5893300
